ros_buildfarm/templates/misc/rosdistro_cache_job.xml.em

```diff
@@ -79,6 +79,9 @@
         ' --cidfile=$WORKSPACE/docker_generate_rosdistro_cache/docker.cid' +
         ' --net=host' +
         ' -v $WORKSPACE/rosdistro_cache:/tmp/rosdistro_cache' +
+        ' -v $HOME/.ssh/known_hosts:/etc/ssh/ssh_known_hosts:ro' +
+        ' -v $SSH_AUTH_SOCK:/tmp/ssh_auth_sock' +
+        ' -e SSH_AUTH_SOCK=/tmp/ssh_auth_sock' +
```

Comment:
Sorry about the delay. I see the rationale for why those arguments should only be passed conditionally and I'm happy to make that change.
Does this look like a reasonable way to conditionally add or not add these arguments to the docker run invocation?
```
        ' -v $HOME/.ssh/known_hosts:/etc/ssh/ssh_known_hosts:ro' +
        '{}'.format(' -v $SSH_AUTH_SOCK:/tmp/ssh_auth_sock -e SSH_AUTH_SOCK=/tmp/ssh_auth_sock' if git_ssh_credential_id else '') +
        ' rosdistro_cache_generation',
```